kong/tools/timestamp.lua

```diff
@@ -40,31 +40,46 @@ local function get_timetable(now)
   return tt_from_timestamp(timestamp)
 end
 
+
 --- Creates a timestamp table containing time by different precision levels.
 -- @param now (optional) Time to generate timestamps from, if omitted current UTC time will be used
+-- @param tracked_timestamp (optional) Timestamp tracked, pairs of key/value in : {["second"]= true, ["minute"]= true, ["hour"]= true, ["day"]= true, ["month"]= true, ["year"]= true}
 -- @return Timestamp table containing fields/precisions; second, minute, hour, day, month, year
-local function get_timestamps(now)
+local function get_timestamps(now, tracked_timestamp)
   local timetable = get_timetable(now)
+  local tracked_timestamp = tracked_timestamp or {["second"]= true, ["minute"]= true, ["hour"]= true, ["day"]= true, ["month"]= true, ["year"]= true}
   local stamps = {}
 
   timetable.sec = math_floor(timetable.sec)   -- reduce to second precision
-  stamps.second = timetable:timestamp() * 1000
+  if tracked_timestamp["second"] ~= nil  then
+    stamps.second = timetable:timestamp() * 1000
+  end
 
   timetable.sec = 0
-  stamps.minute = timetable:timestamp() * 1000
-
+  if tracked_timestamp["minute"] ~= nil then
+    stamps.minute = timetable:timestamp() * 1000
+  end
```

Comment:
Sorry that I did not have the bandwidth to take a close look at the PR.
This is not the correct way to use luatz I think. Though it may be confusing, the call to timestamp is stateful, the first call to get the second, the second call to get the minute, so on and so forth.